File: Part_I/chapter_01_the_role_of_algorithms_in_computing/selection_sort.js
/**
 * @function
 * 
 * @description
 * Selection sort algorithms
 *
 * @param {} `` – .
 */
 function selectSort() {
    var A = [12, 3, 7, 9, 83, 0, 23];
    console.log(A);

    for (var i = 0; i < A.length - 1; i++) {
        var smallest = A[i],
            index = i;
        for (var j = i + 1; j < A.length; j++) {
            if (smallest > A[j]) {
                smallest = A[j];
                index = j;
            }
        }
        A[index] = A[i];
        A[i] = smallest;
    }

    console.log(A);
 }

 selectSort();

File: Part_I/chapter_01_the_role_of_algorithms_in_computing/selection_sort.py
#!/usr/bin/python
#Filename: selection_sort.py


def select_sort():
    A = [12, 3, 7, 9, 83, 0, 23]
    bound = len(A)
    print A

    for i in xrange(0, bound - 1):
        smallest = A[i]
        index = i
        for j in xrange(i + 1, bound):
            if smallest > A[j]:
                smallest = A[j]
                index = j
        A[index] = A[i]
        A[i] = smallest
    print A

if __name__ == '__main__':
    select_sort()

# i = 0
# while i < bound - 1:
#     smallest = A[i]
#     index = i
#     j = i + 1
#     while j < bound:
#         if smallest > A[j]:
#             smallest = A[j]
#             index = j
#         j = j + 1
#     A[index] = A[i]
#     A[i] = smallest
#     i = i + 1
